File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/domain/pullrequests/contracts/PullRequestFetcher.java
package com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.contracts;

import com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.PullRequestEntity;
import com.duxtinto.intellij.plugin.github.codereviews.net.pullrequests.PullRequestQueryParameters;

import javax.annotation.Nonnull;
import java.io.IOException;
import java.util.List;

public interface PullRequestFetcher {
    List<PullRequestEntity> fetchForRepository(@Nonnull String userName, @Nonnull String repoName, PullRequestQueryParameters parameters) throws IOException;
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/net/pullrequests/apiv3/PullRequestFetcher.kt
package com.duxtinto.intellij.plugin.github.codereviews.net.pullrequests.apiv3

import com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.PullRequestDomainContract
import com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.PullRequestEntity
import com.duxtinto.intellij.plugin.github.codereviews.net.Contract
import java.io.IOException
import javax.inject.Inject

class PullRequestFetcher
    @Inject
    constructor(private val apiLoader: Contract.PullRequest.Loader)
    : PullRequestDomainContract.Fetcher {

    override fun fetchOneForRepository(userName: String, repoName: String, id: Long): PullRequestEntity {
        return apiLoader.loadOne(userName, repoName, id.toInt())
    }

    @Throws(IOException::class)
    override fun fetchAllForRepository(userName: String, repoName: String, parameters: PullRequestQueryParameters): List<PullRequestEntity> {
        return apiLoader.loadAll(userName, repoName, parameters)
    }
}
